avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -71,4 +78,35 @@ protected Object deserialize(byte[] payload) throws SerializationException {
       throw new SerializationException("Error deserializing Avro message for id " + id, e);
     }
   }
+
+  private DatumReader getDatumReader(Schema writerSchema) {
+    if (useSpecificAvroReader) {
+      return new SpecificDatumReader(writerSchema, getReaderSchema(writerSchema));
+    } else {
+      return new GenericDatumReader(writerSchema);
+    }
+  }
+
+  private Schema getReaderSchema(Schema writerSchema) {
+    Schema readerSchema = targetSchemaCache.get(writerSchema);
+    if (readerSchema == null) {
+      Class targetClass = SpecificData.get().getClass(writerSchema);
+      if (targetClass != null) {
+        try {
+          readerSchema = ((SpecificRecord) targetClass.newInstance()).getSchema();
```

Comment:
You can but then we'd have to use reflection and more code to actually call this. So this seemed a cleaner option. 

Given we're caching the result and only doing it once simpler code is probably better?